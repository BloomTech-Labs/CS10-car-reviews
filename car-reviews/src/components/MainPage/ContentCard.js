import React, { Fragment } from 'react';
import './ContentCard.css';
import { 
    Card,
    CardBody,
    CardTitle,
    CardText,
    CardImg,
    CardFooter
} from 'reactstrap';

// * TODO: Handle dynamic image sizing
// * TODO: Link out to reviewer's page under review
// * TODO: Add the actual star rating
// * TODO: Distinguish between clicking to view the review vs clicking to view the reviewer. All clicks open the review modal at present
const ContentCard = ({ content, cardType }) => {
    console.log(cardType);
    return(
        <Card className='contentCard'>
            <CardImg 
                // className='contentCard__image'
                top
                width="100%"
                src="https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180" 
            />
            <CardBody>
                { /* here the cardType determines what is conditionally rendered */ }
                { cardType !==  'reviewer' ?  <CardTitle>star rating</CardTitle> : <Fragment /> }
                { cardType !==  'reviewer' ?  <CardTitle>{`${content.year} ${content.make} ${content.model}`}</CardTitle> : <Fragment /> }
                { cardType !==  'reviewer' ?  <CardText>{content.edition}</CardText> : <Fragment /> }
                { cardType === 'featured_review' || 'reviewer' ?  <CardText>{content.username}</CardText> : <Fragment /> }
            </CardBody>
        </Card>
    )
}

export default ContentCard;

// { // here if the type is either a featured review or a popular car it will render the star rating
//     cardType === 'featured_review' || 'popular_car' ?  <p>star rating</p> : <Fragment /> 
// }
// {
//     cardType === 'featured_review' || 'popular_car' ?  <p>car model</p> : <Fragment /> 
// }
// {
//     cardType === 'featured_review' || 'popular_car' ?  <p>car edition</p> : <Fragment /> 
// }
// {
//     cardType === 'featured_review' || 'reviewer' ?  <p>reviewer username</p> : <Fragment /> 
// }