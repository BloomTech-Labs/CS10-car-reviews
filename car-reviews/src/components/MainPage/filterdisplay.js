import React from 'react'
import Select from 'react-select'

export default () => {
  const uniqueItems = Array.From(new Set(list));
  return (
    <Select
        isMulti
        closeMenuOnSelect={false}
        name="colors"
        options={uniqueItems}
        className="basic-multi-select"
        classNamePrefix="select"
        placeholder
    />
  )
}
