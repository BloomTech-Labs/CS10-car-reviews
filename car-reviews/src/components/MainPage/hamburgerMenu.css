#hamburgerMenu {
  display: flex;
  justify-content: flex-end;
}

/* Individual item */
.bm-item {
  display: flex;

  /* Our sidebar item styling */
  text-decoration: none;
  margin-bottom: 10px;
  color: #d1d1d1;
  transition: color 0.2s;
}

/* Change color on hover */
.bm-item:hover {
  color: white;
}

/* Position and sizing of burger button */
.bm-burger-button {
  position: relative;
  width: 36px;
  height: 30px;
  right: 10px;
  top: 20px;
}

/* Color/shape of burger icon bars */
.bm-burger-bars {
  background: #373a47;
}

/* Position and sizing of clickable cross button */
.bm-cross-button {
  height: 24px;
  width: 24px;
}

/* Color/shape of close button cross */
.bm-cross {
  background: #bdc3c7;
}

/* General sidebar styles */
.bm-menu {
  background: #373a47;
  padding: 2.5em 1.5em 0;
  font-size: 1.2em;
}

/* Morph shape necessary with bubble or elastic */
.bm-morph-shape {
  fill: #373a47;
}

/* Wrapper for item list */
.bm-item-list {
  color: #b8b7ad;
}

/* Styling of overlay */
.bm-overlay {
  background: rgba(0, 0, 0, 0.3);
}