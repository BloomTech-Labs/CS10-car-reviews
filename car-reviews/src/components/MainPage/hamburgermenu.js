import React, { Component, Fragment } from 'react';
import { slide as Menu } from 'react-burger-menu';
import AuthService from '../Auth/authservice';

const Auth = new AuthService();

class HamburgerMenu extends Component {
  handleSignOut = () => {
    Auth.logout();
  }

  render(){
    return (
      <Menu {...this.props}>
        <a className="navbar-item" href="/">
          Home
        </a>
  
        {this.props.isLoggedIn ?
        <a className="navbar-item" href="/MyReviews">
          My Reviews
        </a> : <Fragment />
        }

        {this.props.isLoggedIn ?
        <a className="navbar-item" href="/Billing">
          Billing
        </a> : <Fragment />
        }

        {this.props.isLoggedIn ?
        <a className="navbar-item" href="/Settings">
          Settings
        </a> : <Fragment />
        }

        {!this.props.isLoggedIn ?
          <a className='navbar-item' href="/login">Sign In</a> :
          <a className="navbar-item" href="/" onClick={this.handleSignOut}>Sign Out</a>
        }
      </Menu>
    );
  }
};

export default HamburgerMenu;
