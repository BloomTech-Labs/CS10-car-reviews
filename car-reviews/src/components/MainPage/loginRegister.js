import React, { Component, Fragment } from 'react';
import axios from 'axios'
import {
    Col,
    Label,
    Button,
    Alert
} from 'reactstrap'; 
import './LoginRegister.css';
import { Redirect } from 'react-router-dom';


// * TODO: Check if the user already has a valid JWT when they first visit the page and when they navigate to login/signup
// * TODO: Style login forms
// ** OPTIONAL: Create multiple alerts for all the different possible errors (no email, no password, both, etc.)
class LoginRegister extends Component {
    state = {
        login: {
            email: '',
            password: ''
        },
        register: {
            fullname: '',
            username: '',
            email: '',
            password: ''
        },
        redirect: {
            status:false
        },
        alerts: {
            login: false,
            register: false
        }
    }
    handleUpdateForms = (type, field) => (event) => {
        const newState = Object.assign({}, this.state);
        newState[type][field] = event.target.value;
        this.setState(newState);
    }
    handleSubmitForm = (formType) => (event) => {
        event.preventDefault();
        const requestURL = `https://lambda-car-reviews.herokuapp.com/auth/${formType}`;
        const localRequests = `http://localhost:3001/auth/${formType}`
        const userForm = Object.assign({}, this.state[formType]);
        console.log(userForm);
        axios.post(localRequests, userForm)
            .then(response => {
                // removes the alert if it's present
                if (this.state.alerts[formType]) this.handleAlerts(formType);
                // when the user successfully logs in/registers they are issued a JWT that is saved in storage with the key 'jwt'
                localStorage.setItem('jwt', response.data.JWT);
                // here the login status of the user is changed to 'true' when the login/register is successful
                this.setState({
                    login: {
                        email: '',
                        password: ''
                    },
                    register: {
                        fullname: '',
                        username: '',
                        email: '',
                        password: ''
                    },
                    redirect: {
                        status:true
                    }
                })
            })
            .catch(err => {
                if (!this.state.alerts[formType]) this.handleAlerts(formType);
                console.warn(err);
            });
    }

    handleAlerts = (type) => {
        const newState = Object.assign({}, this.state);
        newState.alerts[type] = !this.state.alerts[type];
        this.setState(newState);
    }

    handleRedirect =() => {
        if(this.state.redirect.status){
          return  <Redirect to='/'  />
        } else {
           return <div className="login-container">
            <Col sm="6">
                        <form onSubmit={this.handleSubmitForm('login')}>
                        <Label>Login Please!</Label>
                            <input 
                                value={this.state.login.email} 
                                placeholder='Enter your email...' 
                                onChange={this.handleUpdateForms('login', 'email')}     
                            />
                            <input 
                                // * NOTE: Hide characters
                                type='password'
                                value={this.state.login.password} 
                                placeholder='Enter your password...' 
                                onChange={this.handleUpdateForms('login', 'password')}   
                            />
                            <Button type='submit' color ="primary">Login</Button>
                            <Alert isOpen={this.state.alerts.login} color='danger'>Incorrect email and/or password, please try again</Alert>
                        </form>
                    </Col>
               
                
                
                
                <Col sm="6">
                    {/* Right pane: Login */}
                    { /* * TODO: Add logic to hide register fields until a button is clicked */}
                    <form onSubmit={this.handleSubmitForm('register')}>
                    <Label>Register if you do not have an account yet!</Label>
                        <input 
                            value={this.state.register.fullname} 
                            placeholder='Enter your fullname...' 
                            onChange={this.handleUpdateForms('register', 'fullname')}   
                        />
                        <input 
                            // * NOTE: Hide characters
                            value={this.state.register.username} 
                            placeholder='Enter your username...' 
                            onChange={this.handleUpdateForms('register', 'username')}   
                        />
                        <input 
                            value={this.state.register.email} 
                            placeholder='Enter your email...' 
                            onChange={(this.handleUpdateForms('register', 'email'))}   
                        />
                        <input 
                            // * NOTE: Hide characters
                            value={this.state.register.password} 
                            type='password'
                            placeholder='Enter your password...' 
                            onChange={this.handleUpdateForms('register', 'password')}   
                        />
                         <Button color="primary">Register</Button>
                         <Alert isOpen={this.state.alerts.register} color='danger'>There was an error registering you, please check your credentials and try again</Alert>
                    </form>
                </Col>
                </div>
        }
        
    }
    render(){
        return(
            <Fragment>
               {this.handleRedirect()} 
            </Fragment>
        )
    }
}

export default LoginRegister;