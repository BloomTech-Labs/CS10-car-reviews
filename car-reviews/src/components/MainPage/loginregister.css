.login-container{
   display: flex;
}

.col-sm-4{
    display: flex; 
    /* border: 3px solid blue;  */
    justify-content: center;
    margin-top: 245px;
    margin-bottom: 245px;
    /* height: 350px; */
    height: 100%;
    /* background-color: #B4B6BA;
    opacity: 0.5;
    filter: alpha(opacity=50); */
}

.login-mid {
    font-size: 1.5em;
}

Input {
    /* border: 1px solid purple; */
    display: block;
    margin-top: 20px;
    
}

.general-button {
    margin: 10px;
    width: 90%;
    background-color: white;
    color: #77A6F7;
    border: none !important;
}
.general-button:hover {
    background-color: #2F77F3;
    color: white;
}