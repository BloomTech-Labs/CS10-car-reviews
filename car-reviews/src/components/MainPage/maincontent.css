.main-content-container {
    width: 80%;
    margin-right: 100;
    
}

.main-content {
    width: 80%;
    display: flex;
    justify-content: center;
    margin-right: 1000;
}