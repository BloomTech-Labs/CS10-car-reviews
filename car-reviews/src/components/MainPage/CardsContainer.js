import React, { Component, Fragment } from 'react';
import './CardsContainer.css';
import ContentCard from './ContentCard';

class CardsContainer extends Component {
    handleMappingContent = () => {
        return this.props.content.map(card => {
            return <ContentCard content={card} cardType={this.props.cardType}/>
        })
    }

    render(){
        const { header } = this.props;
        return(
            <div className='cardsContainer__main'>
                <h3>{header}</h3>

                <div className='cardsContainer'>
                    {this.handleMappingContent()}
                </div>
            </div>
        )
    }
}

export default CardsContainer;