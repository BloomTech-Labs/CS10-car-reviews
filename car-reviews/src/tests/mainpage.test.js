import MainPage from '../components/MainPage/mainpage';

describe('MainPage', () => {
  const shallowMainPage = () => shallow(<MainPage />);

  // All tests will go here
  it('renders MainPage component correctly', () => {
    expect(shallowMainPage).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const divs = shallowMainPage().find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  it('always renders a SearchBar', () => {
    const searchbar = shallowMainPage().find('Searchbar');
    expect(searchbar.length).toBe(1);
  });

  describe('SearchBar component should receive exactly two props', () => {
    const searchbar = shallowMainPage().find('Searchbar');
    expect(Object.keys(searchbar.props()).length).toBe(2);
  });

  it('always renders <MainContent/>', () => {
    const maincontent = shallowMainPage().find('MainContent');

    expect(maincontent.length).toBe(1);
  });

  it('<MainContent/> should receive isLoggedIn prop only', () => {
    const maincontent = shallowMainPage().find('MainContent');

    expect(Object.keys(maincontent.props()).length).toBe(1);
  });

  it('always renders two a tags', () => {
    expect(shallowMainPage().find('a').length).toBe(2);
  });

  it('href link in both a tags to be defined', () => {
    expect(
      shallowMainPage()
        .find('a')
        .at(0)
        .props().href
    ).toBeDefined();
    expect(
      shallowMainPage()
        .find('a')
        .at(1)
        .props().href
    ).toBeDefined();
  });

  it('localStorage.getItem called with jwt', () => {
    expect(localStorage.getItem).toBeCalledWith('jwt');
  });

  describe('Testing state of MainPage', () => {
    it('isLoggedIn state is defaulted to false', () => {
      expect(shallowMainPage().state('isLoggedIn')).toBeFalsy();
    });
  });
});
