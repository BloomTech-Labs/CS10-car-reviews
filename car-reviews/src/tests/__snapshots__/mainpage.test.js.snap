// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders MainPage component correctly 1`] = `[Function]`;
